### PR TITLE
feat: add AnalyticsError type for setting fetch failed.

### PR DESCRIPTION
**Pull request by didiergarcia**
Adding AnalyticsError.settingsFetchFailed(Error) type.

Wrap all the errors that occur while fetching settings using the new type.

### CODE REVIEW

**Review by alanjcharles — APPROVED** [review]

File: Sources/Segment/Utilities/Networking/HTTPClient.swift [hunk 1]
```diff
@@ -63,7 +63,7 @@ public class HTTPClient {
 
         let dataTask = session.uploadTask(with: urlRequest, fromFile: batch) { [weak self] (data, response, error) in
             guard let self else { return }
-            handleResponse(data: data, response: response, error: error, completion: completion)
+            handleResponse(data: data, response: response, error: error, urlRequest: urlRequest, completion: completion)
```
Comment (bsneed) [inline Sources/Segment/Utilities/Networking/HTTPClient.swift]: if we just need the url, lets just pass the url instead of the whole request.

**Review by bsneed — APPROVED** [review]

**Review by alanjcharles — APPROVED** [review]
lgtm!

**Comment by alanjcharles** [comment]
although I am not sure what's going on with the linux test CC @bsneed

**Comment by bsneed** [comment]
looks like that disk image is now gone; can explore fixing it later.

File: Sources/Segment/Utilities/Networking/HTTPClient.swift (hunk 1 above)
Comment (didiergarcia) [reply Sources/Segment/Utilities/Networking/HTTPClient.swift]: Done

**Pull request merged**